app/routines/admin/search_security_log.rb

```diff
@@ -0,0 +1,153 @@
+# Routine for searching the Security Log by admins
+#
+# Caller provides a query and some options.  The query follows the rules of
+# https://github.com/bruce/keyword_search , e.g.:
+#
+#   "user:jps,richb" --> returns the security log for the "jps" and "richb" users
+#   "application:Tutor" --> returns the security log for the Tutor application
```

Comment:
Query by IP is in... I'll see what I can do about time ranges, but that's hard...